ethcore/engines/authority-round/src/lib.rs

```diff
@@ -1116,6 +1126,56 @@ impl AuthorityRound {
 		let tx_request = TransactionRequest::call(contract_addr, data).gas_price(U256::zero()).nonce(nonce);
 		Ok(vec![full_client.create_transaction(tx_request)?])
 	}
+
+	fn run_posdao(&self, block: &ExecutedBlock, nonce: Option<U256>) -> Result<Vec<SignedTransaction>, Error> {
+		// Skip the rest of the function unless there has been a transition to POSDAO AuRa.
+		if self.posdao_transition.map_or(true, |posdao_block| block.header.number() < posdao_block) {
+			trace!(target: "engine", "Skipping calls to POSDAO randomness and validator set contracts");
+			return Ok(Vec::new());
+		}
+
+		let opt_signer = self.signer.read();
+		let signer = match opt_signer.as_ref() {
+			Some(signer) => signer,
+			None => return Ok(Vec::new()), // We are not a validator, so we shouldn't call the contracts.
+		};
+		let our_addr = signer.address();
+		let client = self.client.read().as_ref().and_then(|weak| weak.upgrade()).ok_or_else(|| {
```

Comment:
should it use the newly added [`upgrade_client_or`](https://github.com/paritytech/parity-ethereum/pull/11408/files#diff-79ef4a3dce58a9d303d49c5aee781e17R1124) method?